nimare/meta/cbma/ale.py

```diff
@@ -584,25 +632,23 @@ def _fit(self, dataset):
         )
 
         # Determine bins for null distribution histogram
-        max_ma_values = np.max(ma_values, axis=1)
+        max_ma_values = ma_values.max(axis=[1, 2, 3]).todense()
+
         max_poss_ale = self._compute_summarystat_est(max_ma_values)
         self.null_distributions_["histogram_bins"] = np.round(
             np.arange(0, max_poss_ale + 0.001, 0.0001), 4
         )
 
         stat_values = self._compute_summarystat_est(ma_values)
 
+        del ma_values
+
         iter_df = self.inputs_["coordinates"].copy()
         rand_idx = np.random.choice(self.xyz.shape[0], size=(iter_df.shape[0], self.n_iters))
         rand_xyz = self.xyz[rand_idx, :]
         iter_xyzs = np.split(rand_xyz, rand_xyz.shape[1], axis=1)
 
-        perm_scale_values = np.memmap(
-            self.memmap_filenames[1],
-            dtype=stat_values.dtype,
-            mode="w+",
-            shape=(self.n_iters, stat_values.shape[0]),
-        )
+        perm_scale_values = np.zeros((self.n_iters, stat_values.shape[0]), dtype=stat_values.dtype)
```

Comment:
I think we're going to need to profile this.

Reply:
Do you think we should keep the memap array here? 
I profiled the computation time is faster and the memory usage is not that different.

Reply:
```python
from nimare.dataset import Dataset
from nimare.meta.cbma.ale import SCALE
from nimare.utils import get_resource_path, vox2mm

dset_file = os.path.join(get_resource_path(), "nidm_pain_dset.json")
dset = Dataset(dset_file)

xyz = vox2mm(
    np.vstack(np.where(dset.masker.mask_img.get_fdata())).T,
    dset.masker.mask_img.affine,
)

meta = SCALE(xyz=xyz, n_iters=10)
results = meta.fit(dset)

```

This is what I got for 10 iterations:

- `master`:
    - 10/10 [00:03<00:00,  2.95it/s]
    - 228483/228483 [04:32<00:00, 839.13it/s]
    - Current memory usage is 18.818837MB; Peak was 231.680765MB
    - Computation time: 276.62519907951355

- `ale-sparse`:
    - 10/10 [00:06<00:00,  1.65it/s]
    - 228483/228483 [04:11<00:00, 908.33it/s]
    - Current memory usage is 29.558014MB; Peak was 198.31731MB
    - Computation time: 260.77495074272156